java/fury-core/src/main/java/org/apache/fury/serializer/ObjectSerializer.java

```diff
@@ -228,6 +229,65 @@ public void write(MemoryBuffer buffer, T value) {
     writeContainerFields(buffer, value, fury, refResolver, classResolver);
   }
 
+  @Override
+  public T copy(T originObj) {
+    if (isRecord) {
```

Comment:
We need to implement copy for org.apache.fury.builder.Generated.GeneratedSerializer, CompatibleSerializer and MetaSharedSerializer too. Perhaps we need a new base class for those serializer